rbx_types/src/tags.rs

```diff
@@ -0,0 +1,108 @@
+use std::convert::TryFrom;
+use std::str::FromStr;
+use std::string::FromUtf8Error;
+
+#[derive(Default, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(transparent)
+)]
+pub struct Tags {
+    tag_list: Vec<String>,
+}
+
+impl Tags {
+    pub fn new() -> Tags {
+        Self {
+            tag_list: Vec::new(),
+        }
+    }
+
+    pub fn as_vec(&self) -> &Vec<String> {
+        &self.tag_list
+    }
+
+    pub fn as_mut_vec(&mut self) -> &mut Vec<String> {
+        &mut self.tag_list
+    }
+}
+
+impl From<Vec<String>> for Tags {
+    fn from(tag_list: Vec<String>) -> Tags {
+        Self { tag_list }
+    }
+}
+
+impl From<Vec<&str>> for Tags {
+    fn from(tag_list: Vec<&str>) -> Tags {
+        Self {
+            tag_list: tag_list
+                .iter()
+                .map(|tag_name| String::from_str(tag_name).unwrap())
+                .collect(),
+        }
+    }
+}
```

Comment:
Not sure we need this; users can convert to `Vec<String>` first themselves:

```rust
tags.into_iter().map(String::from).collect()
```

If we wanted this, we would be better off implementing `FromIterator` to get a bunch of containers for free.